### DISCUSSION TITLE
tailwind.config.js works but tailwind.config.ts doesnt in create-react-app

### DISCUSSION THREAD

**Discussion by Prasad-Katkade** [tailwindlabs/tailwindcss#12512] [Help] [answered]
I've setup a monorepo with vercel turborepo lib and in one of my react project (typescript based) i am trying to install tailwind 
when i follow the installation guide given at [Install Tailwind CSS with Create React App](https://tailwindcss.com/docs/guides/create-react-app) it works perfectly fine but **as soon as i change the type of tailwind.config.js to tailwind.config.ts it stops working**

I want to setup a shared tailwind config for my apps and common ui components. For UI components it is working but for my react app it is not.

**here is the content of my tailwind.config.js, it is the default config given at installation guide**
```
/** @type {import('tailwindcss').Config} */
// eslint-disable-next-line no-undef
module.exports = {
  content: [
    "./src/**/*.{js,jsx,ts,tsx}",
  ],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

**here is the content of my tailwind.config.ts file**
```
import type { Config } from "tailwindcss";

const config: Config = {
  content: [
    "./src/**/*.{js,jsx,ts,tsx}",
  ],
  theme: {
    extend: {},
  },
  plugins: [],
};
export default config;
```
Here is my configuration - 

1] OS - Windows 
2] React version - 17.0.2
3] Node version - 18.18.0
4] Tailwindcss version - 3.3.5

I am not sure what is causing this issue. Your Help is really appreciated !

**Comment by wongjn** [comment, marked as answer]
`react-scripts` is causing the issue. in `react-scripts/config/webpack.config.js`, [it only checks for `tailwind.config.js` for whether to include Tailwind in the build chain](https://github.com/facebook/create-react-app/blob/main/packages/react-scripts/config/webpack.config.js#L71-L74):

```js
// Check if Tailwind config exists
const useTailwind = fs.existsSync(
  path.join(paths.appPath, 'tailwind.config.js')
);
```
To work around this, you could consider:
- [Ejecting the configuration](https://create-react-app.dev/docs/available-scripts/#npm-run-eject) to customize it for your needs.
- Using a more modern build tool, like [Vite](https://tailwindcss.com/docs/guides/vite).

**Reply by Prasad-Katkade** [reply]
Thanks @wongjn
